### PR TITLE
feat(agent): dagent ipc health

### PR DESCRIPTION
**Pull request by robot9706**
Implemented a unix socket based health check for dagent.

### CODE REVIEW

File: golang/internal/health/server.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+}
+
+func SetHealthGRPCStatus(connected bool) {
+	health.Connected = connected
+}
+
+func Serve(ctx context.Context) error {
+	_, err := os.Stat(socketPath)
+	if err == nil {
+		err = os.RemoveAll(socketPath)
```
Comment (nandor-magyar) [inline golang/internal/health/server.go:53]: The first error is swallowed and only the remove error is printed. Is it intentional/ fail-safe?

File: golang/internal/health/types.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package health
+
+const (
+	socketType = "unix"
+	socketPath = "/tmp/agenthealth.sock"
```
Comment (nandor-magyar) [inline golang/internal/health/types.go:5]: A better path would be: path.Join(os.TempDir(),"dyrectorio","agenthealth.sock"), pay attention to file existence privilege error handling too.

**Review by nandor-magyar — COMMENTED** [review]
Minor details, look neat otherwise. :sunglasses:

**Comment by nandor-magyar** [comment]
Some small tests would be nice.

File: golang/internal/grpc/grpc.go [hunk 3]
```diff
@@ -275,22 +282,24 @@ func grpcLoop(
...
 		if err != nil {
-			log.Error().Stack().Err(err).Msg("Cannot receive stream")
 			grpcConn.Client = nil
+			health.SetHealthGRPCStatus(false)
+
+			if err == io.EOF {
+				log.Info().Msg("End of stream")
+			} else {
+				log.Error().Stack().Err(err).Msg("Cannot receive stream")
+				// TODO replace the line above with an error status code check and terminate dagent accordingly
```
Comment (polaroi8d) [inline golang/internal/grpc/grpc.go:299]: Do we still need this comment, @nandor-magyar? Is this comment from the `GitLab` repository? If yes, what is blocking us from solving this issue?

**Review by polaroi8d — APPROVED** [review]
Minor comment,otherwise informal LGTM.

File: golang/internal/grpc/grpc.go (hunk 3 above)
Comment (nandor-magyar) [reply golang/internal/grpc/grpc.go:299]: I don't remember exactly, but if EOF means that the remote connection is closed then agent should exit as well, other errors should be investigated what is possible to get here.

**Review by nandor-magyar — APPROVED** [review]

File: golang/internal/grpc/grpc.go (hunk 3 above)
Comment (nandor-magyar) [reply golang/internal/grpc/grpc.go:299]: I'll create an issue.

**Pull request merged**